src/sas/sascalc/dataloader/loader.py

```diff
@@ -61,31 +61,32 @@ def load(self, path, format=None):
         :param format: explicit extension, to force the use
             of a particular reader
 
-        Defaults to the ascii (multi-column) reader
-        if no reader was registered for the file's
-        extension.
+        Defaults to the ascii (multi-column), cansas XML, and cansas NeXuS
+        readers if no reader was registered for the file's extension.
         """
         try:
             return super(Registry, self).load(path, format=format)
-        except:
-            try:
-                # No reader was found. Default to the ascii reader.
-                ascii_loader = ascii_reader.Reader()
-                return ascii_loader.read(path)
-            except:
-                try:
-                    cansas_loader = cansas_reader.Reader()
-                    return cansas_loader.read(path)
-                except:
-                    try:
-                        cansas_nexus_loader = cansas_reader_HDF5.Reader()
-                        return cansas_nexus_loader.read(path)
-                    except:
-                        msg = "\n\tUnknown data format: %s.\n" % path
-                        msg += "\tThe file is not a known format for SasView. "
-                        msg += "The most common formats are multi-column "
-                        msg += "ASCII, CanSAS XML, andCanSAS NeXuS."
-                        raise Exception(msg)
+        except Exception:
+            pass # try the ASCII reader
+        try:
+            ascii_loader = ascii_reader.Reader()
+            return ascii_loader.read(path)
+        except Exception:
+            pass # try the cansas XML reader
+        try:
+            cansas_loader = cansas_reader.Reader()
+            return cansas_loader.read(path)
+        except Exception:
+            pass # try the cansas NeXuS reader
+        try:
+            cansas_nexus_loader = cansas_reader_HDF5.Reader()
+            return cansas_nexus_loader.read(path)
+        except Exception:
+            # No known reader available. Give up and throw an error
+            msg = "\n\tUnknown data format: %s.\n\tThe file is not a " % path
+            msg += "known format for SasView. The most common formats are "
+            msg += "multi-column ASCII, CanSAS XML, and CanSAS NeXuS."
+            raise Exception(msg)
```

Comment:
Prefer RuntimeError(msg) over Exception(msg) for a non-specific error.  Looking at the standard errors available [https://docs.python.org/2.7/library/exceptions.html], nothing else seems to apply.  Don't use IOError, since that usually is defined using IOError(errno=code, strerror=msg), and refers to a specific OS defined error code.  You could instead make a specific error for sasview loaders which is a subclass of Exception.

It might be useful to display a reader-specific error message instead of just "Unknown data format".  You could use magic strings at the start of the file to identify cansas files, and otherwise try the ascii reader:

    HDF magic string: http://www.digitalpreservation.gov/formats/fdd/fdd000229.shtml#sign
    XML: "<?xml"  or "<SASroot"

Maybe save the traceback from the default reader, and include that in your RuntimeException if the alternate reader does not work.